### PR TITLE
chore: ica submodules minor improvements and cleanup

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

The following PR includes some residual follow up tasks from #541, namely:

- Updating GenesisState protobuf types to include `(gogoproto.nullable) = false` in favour of pointer types
- Updating GenesisState functions and tests to accommodate value types
- Updating module.go `ExportGenesis` to include default genesis types in favour of `nil` pointers
- Removing `GetAllPorts` unused function from `host` submodule
- Moving `ValidateBasic` check in relay.go to avoid unnecessary loop iterations

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/27-interchain-accounts/module.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ func (AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
 
 // ValidateGenesis performs genesis state validation for the IBC interchain acounts module
 func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, config client.TxEncodingConfig, bz json.RawMessage) error {
-	return nil
+	return nil // TODO: https://github.com/cosmos/ibc-go/issues/535
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/module.go:59]: I figured implementing `ValidateGenesis` should be kept to it's own PR entirely. Wanted to keep this one pretty concise.

File: modules/apps/27-interchain-accounts/host/keeper/relay.go [hunk 2]
```diff
@@ -32,16 +32,14 @@ func (k Keeper) executeTx(ctx sdk.Context, sourcePort, destPort, destChannel str
 		return err
 	}
 
+	// CacheContext returns a new context with the multi-store branched into a cached storage object
+	// writeCache is called only if all msgs succeed, performing state transitions atomically
+	cacheCtx, writeCache := ctx.CacheContext()
```
Comment (seantking) [inline modules/apps/27-interchain-accounts/host/keeper/relay.go:37]: Why move? Just wondering

Comment (seantking) [reply modules/apps/27-interchain-accounts/host/keeper/relay.go:37]: Ah sorry I see its the same for loop now. Diff confused me :)

**Review by seantking — APPROVED** [review]

**Review by colin-axner — APPROVED** [review]
Amazing, thanks! I love how easy this was to review. Thanks for keeping it well scoped
